config/menus/maps.cfg

```diff
@@ -213,455 +91,681 @@ mutsvar = [
+mapselection = [
     ui_list [
-        ui_strut 70 1
         ui_list [
-            ui_list [
+            local j
+            j = 0
+            looplist i (sublist $maplist $mapindex 8) [
+                isfav = (!= (listfind j $mapfavs [result (=s $j $i)]) -1)
+                ui_strut 0.2
                 ui_list [
-                    ui_strut 3 1
+                    if (=s $chosenmap $i) [ui_background 0x444444] [ui_background 0x222222]
                     ui_list [
-                        ui_strut 38 1
-                        if (< $last_mode 0) [
-                            ui_font "emphasis" [ ui_text "Game Select" ]
-                            ui_text "Please select a mode and map to continue"
-                        ] [
-                            gname = (gamename $selected_mode $selected_mutators 0 32)
-                            if (&& (stringlen $current_map) (= $selected_mode $modeidxdemo)) [
-                                dinfo = (demoscan (format "%1.dmo" $map_pth))
-                                dmode = (demoinfo $dinfo 2)
-                                dmuts = (demoinfo $dinfo 3)
-                                dname = (gamename $dmode $dmuts 0 26)
-                                gname = (concat $dname "(demo)")
-                            ]
-                            ui_font "emphasis" [ ui_text $gname ]
-                            ui_list [
-                                if (stringlen $current_map) [
-                                    if (= $selected_mode $modeidxdemo) [
-                                        dinfo = (demoscan (format "%1.dmo" $map_pth))
-                                        dmapname = (demoinfo $dinfo 1)
-                                        ui_text " ^farecorded on "
-                                        ui_text $dmapname
-                                    ] [
-                                        ui_text " ^faselected on "
-                                        ui_text $current_map
-                                    ]
-                                ] [ ui_text "Please select a map to continue" ]
+                        ui_merge 30 [
+                            if (! ($hidemappreviews)) [
+                                ui_image (? (= $mapcategory 2) $i (concatword "maps/" $i)) [] 1.5 1 "textures/emblem"
+                            ] [ 
+                                ui_image "textures/emblem" [] 1.5 1
                             ]
+                            ui_strut 1
+							ui_list [
+								if (!=s $i "<random>") [
+									ui_strut 0.5
+									ui_button (concatword (? ($isfav) "^fb" "") (truncate_string (? (= $mapcategory 2) (substring $i 5) $i) 29)) []
+									ui_font "little" [
+										if (!= $mapcategory 2) [ui_button (getmapmodemuts $i) []] [ui_button "^f[0x808080]sauerbraten" []]
+									]
+									ui_strut 25 1
+								] [
+									ui_strut 0.5
+									ui_button "<Random>"
+									ui_strut 28 1
+								]
+							]
```

Comment:
Needs correct indention.

These are tabs, they should be four spaces per indent.